### PR TITLE
[bug] fix #1379 change "Unknown contact" message to "Anonymous contact"

### LINKED ISSUE #1379: Show anonymous usernames instead of Unknown contact in public chat
### Description

*Type*: Bug

*Summary*: All participants in public chat have "unknown contact" title. It would be easier to distinguish several users if anonymous usernames are shown instead of same Unknown contact title.
img_1906

#### Expected behavior
Anonymous username is shown as a user name in public chat to other participants
#### Actual behavior
Unknown contact  is shown as a user name in public chat to other participants

### Reproduction

- Open Status
- Join public chat as contact A
- Join public chat as contact B
-  as contact A: write a message to public chat
-  as contact B: check the user name shown above the message from A in public chat

### Additional Information
* Status version: 0.9.9
* Operating System: iOS and Android

### PR DESCRIPTION
**Pull request by SHA-2048**
fix #1379 change "Unknown contact" message to "Anonymous contact"

### CODE REVIEW

**Comment by andytudhope** [comment]
Small change, but some might say that you're rapidly becoming our most prolific contributor @mlahia 😉 I like this one, think it is more descriptive and accurate - @jeluard @oskarth thoughts?

**Comment by jeluard** [comment]
@annadanchenko Is this what you expected? It's not clear to me how you can distinguish users after this patch.

**Comment by goranjovic** [comment]
@mlahia Hey, great to see you gaining momentum with names! Incidentally, that's the part of the code I started with so extra cool from my perpective  :D

Just wanted to bring some extra clarity to this - all Status users get a generated username, when they first create an account. It is in the following format *adjective adjective animal*, e.g. Cool Bored Panda.

That's done by the function you worked on in the previous PR. The generator fn depends on user's public key, so the same account will always have the same "generated" name, which we also sometimes refer to as "anonymous". Then, a user can also decide to change that and put his or her actual name in which case the generated name is no longer used.

The thing is that these custom chosen usernames are only visible to your immediate contacts, and everyone else should only see the generated name (that's why we sometimes call them anonymous). So, if I send you a message (and we're in each other contacts), you'll see that it arrived from Goran and someone else that it arrived from say Cool Bored Panda.

**Comment by goranjovic** [comment]
TL:DR 
You're supposed to call the function you enchanced in the last PR and pass the senders public key (i.e. whisper identity) - that way those anonymous/generated names will appear for each user their own - not one hardcoded string for everyone.

File: src/status_im/chat/views/message/message.cljs [hunk 1]
```diff
@@ -47,7 +47,7 @@
    incoming-name [:contact-name-by-identity from]]
   (if-let [name (if outgoing
                   (:name current-account)
-                  (or incoming-name "Unknown contact"))]
+                  (or incoming-name "Anonymous contact"))]
```
Comment (goranjovic) [inline src/status_im/chat/views/message/message.cljs]: Replace the hardcoded string with a call to `generate-gfy` - see comment on PR for the context

**Review by goranjovic — CHANGES_REQUESTED** [review]

File: src/status_im/chat/views/message/message.cljs (hunk 1 above)
Comment (SHA-2048) [reply src/status_im/chat/views/message/message.cljs]: Ok I understand know what "anonymous" actually means. Thanks a lot for the detailed explanation!

**Review by goranjovic — APPROVED** [review]

**Review by jeluard — APPROVED** [review]

**Comment by annadanchenko** [comment]
Issue found during testing, please see https://github.com/status-im/status-react/issues/1379#issuecomment-346064292

**Comment by oskarth** [comment]
@mlahia Closing this PR due to inactivity. If you feel like this is in error, please feel free to reopen.

**Pull request closed without merging**